File: tool/table/client/fightbuff.ts
/*
Author: 
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
*/

module config{

let fightbuff_map = null;
export function fightbuff_map_init(config_obj:Object):void{
	fightbuff_map = config_obj["fightbuff_map"];
}


export class Fightbuff extends Object{
	private m_config:any;
	public static m_static_map:Object = new Object();
	constructor(key){
		super();
		this.m_config = fightbuff_map[key];
		for(let i in this.m_config){
			this[i] = this.m_config[i];
		}
		return;
	}
	public static get_Fightbuff(key){
		if(Fightbuff.m_static_map.hasOwnProperty(key) == false){
			Fightbuff.m_static_map[key] = Fightbuff.create_Fightbuff(key);
		}
		return Fightbuff.m_static_map[key];	
	}
	public static create_Fightbuff(key){
		if(fightbuff_map.hasOwnProperty(key)){
			return new Fightbuff(key);
		}
		return null;	
	}
	public static get_cfg_object():Object
	{
		return fightbuff_map;
	}
}
}

File: tool/table/server/fightbuff.py
# -*- coding: utf-8 -*-

'''
Author: Hannibal
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
'''


fightbuff_map = {};
fightbuff_map[1001] = {"id":1001,"name":"强壮","type":"增益","refresh":1,"count":3,"data":[{"prop":"气血增加","value":"1000.0",},],};
fightbuff_map[1002] = {"id":1002,"name":"强力","type":"增益","refresh":1,"count":1,"data":[{"prop":"攻击增加","value":"100.0",},],};


class Fightbuff:
	def __init__(self, key):
		config = fightbuff_map.get(key);
		for k, v in config.items():
			setattr(self, k, v);
		return

def create_Fightbuff(key):
		config = fightbuff_map.get(key);
		if not config:
			return
		return Fightbuff(key)

